tests/integration/pkg/tests/custom/custom_operations_test.go

```diff
@@ -0,0 +1,475 @@
+//go:build operations
+// +build operations
+
+package custom
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+
+	provisioningv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/capr"
+	"github.com/rancher/rancher/pkg/controllers/dashboardapi/settings"
+	"github.com/rancher/rancher/pkg/provisioningv2/kubeconfig"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/rancher/tests/integration/pkg/clients"
+	"github.com/rancher/rancher/tests/integration/pkg/cluster"
+	"github.com/rancher/rancher/tests/integration/pkg/systemdnode"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+func TestCustomEtcdSnapshotOperations(t *testing.T) {
+	clients, err := clients.New()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Nit: could also do this for brevity
```suggestion
	assert.NoError(t, err)
```